src/core/service.c

```diff
@@ -2873,7 +2872,7 @@ static void service_notify_cgroup_empty_event(Unit *u) {
                 /* Fall through */
 
         case SERVICE_START_POST:
-                if (s->pid_file_pathspec) {
+                if (s->pid_file_pathspec && main_pid_good(s) <= 0) {
```

Comment:
This again focuses only on the main process, for which we expect a SIGCHLD event when it dies.

Reply:
well, note that when a PID is "alien" (i.e not PID1's child) then we won't get SIGCHLD for it.

Reply:
Is that even possible for this case (maybe the check is superfluous anyway)?

Reply:
true, this is redundant afaics. if we knew the main pid we'd not watch the PID file